modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -1318,7 +1334,10 @@ private void cancel(UUID nodeId) {
 
                 cleanupRemoteContexts(nodeId);
 
-                remaining.remove(nodeId);
+                IgniteDhtDemandedPartitionsMap rmvd = remaining.remove(nodeId);
+
+                if (rmvd != null)
+                    grp.metrics().cancelRebalancingPartitions(rmvd.size());
```

Comment:
Do we need this method? Can the `reset` method be used? Are we need such callback here at all?